viewing_party/party.py

```diff
@@ -1,23 +1,186 @@
 # ------------- WAVE 1 --------------------
 
+from tests.test_constants import GENRE_1, MOVIE_TITLE_1, RATING_1
+
+
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+
+
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+        return new_movie
+    else: 
+        return None
+
+    
+def add_to_watched(user_data, movie):
+    user_data = {"watched": []}
+    
+    movie = {
+        "title": MOVIE_TITLE_1,
+        "genre": GENRE_1,
+        "rating": RATING_1
+    }
+
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {"watchlist": []}
+    movie = {
+    "title": MOVIE_TITLE_1,
+    "genre": GENRE_1,
+    "rating": RATING_1
+    }
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+
+    
+    for key, value in user_data.items():
+        if key == "watchlist":
+            for i in value:
+                for key, value in i.items():
+                    if value == title:
+                        user_data["watched"].append(title)
+                        user_data["watchlist"].remove(i)
+        else:
+            return user_data
+        
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_total = []
 
+    for i in user_data["watched"]:
+        rating_total.append(i["rating"])
+    if len(rating_total) != 0:
+        return sum(rating_total)/len(rating_total)
+    else: 
+        return 0.0

```

Comment:
We can also use the guard clause pattern here so that the first line of code that executes in this function is a check to see if the watched list is empty, if it is then you can immediately return 0.0

After your guard clause, then you can proceed with the rest of your logic for finding the total.
```python
def get_watched_avg_rating(user_data):
    if not user_data["watched"]: # recall that an empty list is falsy so we can write this check like this for our code to be more pythonic
        return 0.0

    # rest of your logic to find the average goes here